textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexToGramQueryTranslatorTest.java

```diff
@@ -17,133 +17,321 @@ private void printTranslatorResult(String regex) {
-		expectedQuery.operandSet.addAll(Arrays.asList(new String[]{}));
+		String regex = "ab";
+		
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		GramBooleanQuery dnf = GramBooleanQuery.toDNF(exactQuery);
+		GramBooleanQuery simplifiedDNF = GramBooleanQuery.simplifyDNF(dnf);
+		
+		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.OR);
 		
-		Assert.assertTrue(exactQuery.equals(expectedQuery));
+//		printTranslatorResult(regex);
+
+		Assert.assertTrue(simplifiedDNF.equals(expectedQuery));
 	}
 	
 	@Test
 	public void testLiteral3() {
-		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate("abcd");
-		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
-		expectedQuery.operandSet.addAll(Arrays.asList(new String[]{"abc", "bcd"}));
+		String regex = "abcd";
+		
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		GramBooleanQuery dnf = GramBooleanQuery.toDNF(exactQuery);
+		GramBooleanQuery simplifiedDNF = GramBooleanQuery.simplifyDNF(dnf);
 		
-		Assert.assertTrue(exactQuery.equals(expectedQuery));
+		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.OR);
+		GramBooleanQuery expectedAndNode = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
+		expectedAndNode.operandSet.addAll(Arrays.asList(new String[]{"abc", "bcd"}));
+		expectedQuery.subQuerySet.add(expectedAndNode);
+		
+//		printTranslatorResult(regex);
+
+		Assert.assertTrue(simplifiedDNF.equals(expectedQuery));
 	}
 	
 	@Test
 	public void testLiteral4() {
-		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate("ucirvine");
-		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
-		expectedQuery.operandSet.addAll(Arrays.asList(new String[]{"uci", "cir", "irv", "rvi", "vin", "ine"}));
+		String regex = "ucirvine";
+		
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		GramBooleanQuery dnf = GramBooleanQuery.toDNF(exactQuery);
+		GramBooleanQuery simplifiedDNF = GramBooleanQuery.simplifyDNF(dnf);
+
+		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.OR);
+		GramBooleanQuery expectedAndNode = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
+		expectedAndNode.operandSet.addAll(Arrays.asList(new String[]{"uci", "cir", "irv", "rvi", "vin", "ine"}));
+		expectedQuery.subQuerySet.add(expectedAndNode);
 		
-		Assert.assertTrue(exactQuery.equals(expectedQuery));
+//		printTranslatorResult(regex);
+
+		Assert.assertTrue(simplifiedDNF.equals(expectedQuery));
 	}
 	
 	@Test
 	public void testLiteral5() {
-		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate("textdb");
-		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
-		expectedQuery.operandSet.addAll(Arrays.asList(new String[]{"tex", "ext", "xtd", "tdb"}));
+		String regex = "textdb";
```

Comment:
the `ucirvine` and `textdb` are testing the same behavior. You can remove one, or select other conner case.